quinn-proto/src/connection/mod.rs

```diff
@@ -703,6 +704,21 @@ where
             }
 
             let sent = self.populate_packet(space_id, &mut buf, buf_capacity - builder.tag_len);
+
+            // ACK-only packets should only be sent when explicitely allowed
+            // If we write them due to another reason there is a bug somewhere
+            // which leads to one component announcing write readiness and not writing any data.
+            // This degrades performance.
+            // The condition is only checked if the full MTU is available, so that lack of
+            // space in the datagram isn't the reaason for just writing ACKs.
```

Comment:
Fixed some typos, also compacted the comment a bit.

```suggestion
            // ACK-only packets should only be sent when explicitly allowed. If we write them due
            // to any other reason, there is a bug which leads to one component announcing write
            // readiness while not writing any data. This degrades performance. The condition is
            // only checked if the full MTU is available, so that lack of space in the datagram isn't
            // the reason for just writing ACKs.
```